### PR TITLE
Docs: Add light-theme and use users preferred theme

### PR DESCRIPTION
**Pull request by lessp**
I wasn't able to build the complete documentation, but manually checked what I could. 

(disclaimer, I fed the `CSS` to ChatGPT and asked it to extract all the colors into a theme, I then asked it to generate a light theme based on the dark theme colors. From what I could see, the theme looks nice, IMO, of course.)

https://user-images.githubusercontent.com/17602389/234622454-e722a496-6bf8-4f22-811a-6ac66c2cb7c6.mov

Would close: #235

### CODE REVIEW

File: docs/web/site/docs.css [hunk 1]
```diff
@@ -60,6 +60,51 @@
...
+/* Theme */
+
+/* Dark theme (default) */
+:root {
+  --bg-color: #131618;
+  --text-color: #c9d1d9;
+  --pre-bg-color: #1a1f26;
+  --code-bg-color: #2c333b;
+  --border-color: #282828;
+  --h1-text-shadow: black;
```
Comment (lessp) [inline docs/web/site/docs.css:72]: This feels questionable.

Comment (lessp) [reply docs/web/site/docs.css:72]: Removed incorrect references to `h1-text-shadow` in [741a597](https://github.com/aantron/dream/pull/268/commits/741a59739680bca79e0f9939422ead7803f5f3d8)

**Comment by aantron** [comment]
Amazing! It seems fitting to review it on stream tomorrow, then! Thank you!

**Comment by lessp** [comment]
I ran into this issue upon running `opam install --deps-only ./dream.opam --with-test`. Looks like it's unable to find `openssl` in my path?

```
❯ opam install --deps-only ./dream.opam --with-test
The following actions will be performed:
  ∗ install conf-libssl  4
  ∗ install ssl          0.5.13
  ∗ install lwt_ssl      1.2.0
  ∗ install dream-httpaf 1.0.0~alpha2
===== ∗ 4 =====
Do you want to continue? [Y/n] y

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫 
⬇ retrieved ssl.0.5.13  (cached)
⬇ retrieved lwt_ssl.1.2.0  (cached)
⬇ retrieved dream-httpaf.1.0.0~alpha2  (cached)
[ERROR] The compilation of conf-libssl.4 failed at "sh -ex ./homebrew.sh check".

#=== ERROR while compiling conf-libssl.4 ======================================#
# context     2.1.4 | macos/x86_64 | ocaml-base-compiler.4.12.0 | https://opam.ocaml.org#88362781
# path        ~/dev/oss/dream/_opam/.opam-switch/build/conf-libssl.4
# command     ~/.opam/opam-init/hooks/sandbox.sh build sh -ex ./homebrew.sh check
# exit-code   1
# env-file    ~/.opam/log/conf-libssl-46563-ea0c79.env
# output-file ~/.opam/log/conf-libssl-46563-ea0c79.out
### output ###
# ++ brew --prefix openssl
# + brew_pkg_config=/usr/local/opt/openssl@3/lib/pkgconfig
# + case "$1" in
# + test 1 '!=' 1
# + export PKG_CONFIG_PATH=/usr/local/opt/openssl@3/lib/pkgconfig:/Users/ekander/dev/oss/dream/_opam/lib/pkgconfig:
# + PKG_CONFIG_PATH=/usr/local/opt/openssl@3/lib/pkgconfig:/Users/ekander/dev/oss/dream/_opam/lib/pkgconfig:
# + pkg-config --print-errors --exists openssl
# Package openssl was not found in the pkg-config search path.
# Perhaps you should add the directory containing `openssl.pc'
# to the PKG_CONFIG_PATH environment variable
# No package 'openssl' found



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫 
┌─ The following actions failed
│ λ build conf-libssl 4
└─ 
╶─ No changes have been performed
```

**Comment by aantron** [comment]
The docs error is a mismatch during the rewriting of the docs (fixing up odoc's output). Are you on odoc 2.0.2?

The switch has to be less than 5.0.0, because odoc 2.0.2's deps are incompatible with 5.0.0 IIRC. So I use 4.14.1. You shouldn't have to do `--update-invariant` or anything, at least I just start with a 4.14.1 switch.

It looks like you solved the submodule issue.

Not sure what is still left not working. OpenSSL?

**Comment by aantron** [comment]
I better add some notes to CONTRIBUTING.md.

**Comment by lessp** [comment]
> The docs error is a mismatch during the rewriting of the docs (fixing up odoc's output). Are you on odoc 2.0.2?
> 
> 

Ah, right. Let me try that! 🙂

> 
> The switch has to be less than 5.0.0, because odoc 2.0.2's deps are incompatible with 5.0.0 IIRC. So I use 4.14.1. You shouldn't have to do `--update-invariant` or anything, at least I just start with a 4.14.1 switch.
> 
> 

Yeah, I'm not too familiar with Opam, but as far as I could tell, I was using `4.14.1`.

> 
> It looks like you solved the submodule issue.
> 
> 
> 
> Not sure what is still left not working. OpenSSL?

All solved except for the probable odoc mismatch!

**Comment by lessp** [comment]
> The docs error is a mismatch during the rewriting of the docs (fixing up odoc's output). Are you on odoc 2.0.2?

Was definitely not meant as a pique, just thought it'd be helpful to myself and others to document any issues!

**Comment by aantron** [comment]
It's in a pretty neat state. You can see it [live](https://aantron.github.io/dream/). If you have a good idea for where to move the toggle, you can ping this issue or maybe even PR it :) By the way, should the theme button show the icon for the theme you would switch to (as now), or the current theme? The latter seems more intuitive to me, but I'm not sure if there is some kind of usability convention I am not aware of.

**Pull request merged**